tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -82,3 +88,55 @@ void ROOT::Experimental::Detail::RPageSinkBuf::ReleasePage(RPage &page)
 {
    fInnerSink->ReleasePage(page);
 }
+
+namespace {
+   using namespace ROOT::Experimental::Detail;
+   struct RPageZipItem {
+      RPage fPage;
+      RNTupleCompressor fCompressor;
+      RPageStorage::RSealedPage fSealedPage;
+      explicit RPageZipItem(RPage page)
+         : fPage(page), fCompressor(RNTupleCompressor(fPage.GetSize())) {}
+   };
+} // anonymous namespace
+
+void ROOT::Experimental::Detail::RPageSinkBuf::ParallelClusterZip(
+   ROOT::Experimental::NTupleSize_t nEntries)
+{
+   // TODO(max) add timers like in RPageSourceFile::UnzipClusterImpl
+   R__ASSERT(fTaskScheduler);
+   fTaskScheduler->Reset();
+   std::vector<std::vector<RPageZipItem>> zipItems;
```

Comment:
Let's add a comment about the meaning of the two array dimensions

Reply:
Superseded by the revised implementation in e9c6074